lib/boom_notifier.ex

```diff
@@ -1,16 +1,57 @@
 defmodule BoomNotifier do
   @moduledoc false
+
   # Notify the exception to all the defined notifiers
 
   alias BoomNotifier.ErrorStorage
 
+  def walkthrough_notifiers(settings, callback) do
+    case Keyword.get(settings, :notifiers) do
+      nil ->
+        with {:ok, notifier} <- Keyword.fetch(settings, :notifier),
+             {:ok, options} <- Keyword.fetch(settings, :options) do
+          callback.(notifier, options)
+        end
+
+      notifiers_config when is_list(notifiers_config) ->
+        for notifier_config <- notifiers_config do
+          with {:ok, notifier} <- Keyword.fetch(notifier_config, :notifier),
+               {:ok, options} <- Keyword.fetch(notifier_config, :options) do
+            callback.(notifier, options)
+          end
+        end
+    end
+  end
+
   defmacro __using__(config) do
     quote location: :keep do
       use Plug.ErrorHandler
 
       import BoomNotifier
       require Logger
 
+      settings = unquote(config)
+
+      try do
+        walkthrough_notifiers(
+          settings,
+          &if function_exported?(&1, :validate!, 1) do
```

Comment:
Optional: What about making the `validate` callback mandatory? It seems kind of natural to expect that implemented from a notifier.

Reply:
I think this would introduce a breaking change since custom notifiers will be forced to implement the validate callback

Reply:
Anyways, I think it's expected for `0.x.x` version to introduce breaking changes, isn't it?

Reply:
Should it be mandatory then?